quokka_editor_front/src/components/organisms/documentsEditor/EditorPage.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from "react";
+import DocumentsEditor from "./DocumentsEditor";
+import { ClientState } from "../../../types/ot";
+import { useLocation } from "react-router-dom";
+import axios from "axios";
+import { API_URL } from "../../../consts";
+import DocumentTitleUpdate from "./DocumentTitleUpdate";
+import ShareDocument from "./ShareDocument";
+import { DocumentState } from "../../../Redux/documentsSlice";
+
+const initialClient = {
+  lastSyncedRevision: 0,
+  pendingChanges: [],
+  sentChanges: null,
+  documentState: "",
+};
+
+const initialDocument = {
+  title: "",
+  content: " ",
+  id: "",
+  selected: false,
+  shared_role: "READ",
+  shared_by_link: false,
+};
+
+const EditorPage = () => {
+  const [client, setClient] = useState<ClientState>(initialClient);
+  const [document, setDocument] = useState<DocumentState>(initialDocument);
+
+  const location = useLocation();
+  const id = location.pathname.slice(
+    location.pathname.lastIndexOf("/"),
+    location.pathname.length
+  );
+
+  useEffect(() => {
+    axios
+      .get(API_URL + "documents" + id, {
+        headers: { Authorization: sessionStorage.getItem("userToken") },
+      })
+      .then((res) => {
+        setClient({
+          ...client,
+          documentState: JSON.parse(res.data.content).join("\n"),
+          lastSyncedRevision: JSON.parse(res.data.last_revision),
+        });
+        setDocument(res.data);
+      });
+  }, []);
```

Comment:
Why are you still adding the requests and not extracting them to an external function?

Reply:
i will do a new branch where I will move all the requests to a single file